src/routes/class-base-route.php

```diff
@@ -79,80 +94,288 @@ protected function get_route_name() : string {
 
-    return \rest_ensure_response(
-      array(
-        'code' => 400,
-        'message' => esc_html__( 'Unknown error', 'eightshift-forms' ),
-        'data' => $data,
-      )
-    );
+    // If this route requires a filter defined in project, we need to make sure that is defined.
+    if ( ! empty( $required_filter ) && ! has_filter( $required_filter ) ) {
+      throw new Unverified_Request_Exception(
+        $this->rest_response_handler( 'integration-not-used', [ self::MISSING_FILTER => $required_filter ] )->data
+      );
+    }
+
+    $params      = $request->get_query_params();
+    $params      = $this->fix_dot_underscore_replacement( $params );
+    $post_params = $request->get_body_params();
+
+    // Authorized routes need to provide the correct authorization hash to do anything.
+    if ( ! empty( $this->get_authorization_salt() ) ) {
+      $hash = $params[ HMAC::AUTHORIZATION_KEY ] ?? 'invalid-hash';
+      unset( $params[ HMAC::AUTHORIZATION_KEY ] );
+
+      if ( empty( $this->hmac ) || ! $this->hmac->verify_hash( $hash, $params, $this->get_authorization_salt() ) ) {
+        throw new Unverified_Request_Exception(
+          $this->rest_response_handler( 'authorization-invalid' )->data
+        );
+      }
+    }
+
+    // If captcha is used on this route and provided as part of the request, we need to confirm it's true.
+    if ( ! empty( $this->basic_captcha ) && ! $this->basic_captcha->check_captcha_from_request_params( $params ) ) {
+      throw new Unverified_Request_Exception( $this->rest_response_handler( 'wrong-captcha' )->data );
+    }
+
+    // If this route has required parameters, we need to make sure they're all provided.
+    $missing_params = $this->find_required_missing_params( $params );
+    if ( ! empty( $missing_params ) ) {
+      throw new Unverified_Request_Exception(
+        $this->rest_response_handler( 'missing-params', $missing_params )->data
+      );
+    }
+
+    // If this route has required parameters, we need to make sure they're all provided.
+    $missing_post_params = $this->find_required_missing_params( $post_params, true );
+    if ( ! empty( $missing_post_params ) ) {
+      throw new Unverified_Request_Exception(
+        $this->rest_response_handler( 'missing-post-params', $missing_post_params )->data
+      );
+    }
+
+    return $params;
   }
 
   /**
    * Checks if all required parameters are present in request.
    *
    * @param  array $parameters Array of request parameters.
+   * @param  bool  $is_post    (Optional) True if we're checking POST params instead of GET params.
    * @return array Returns array of missing parameters to pass in response.
    */
-  protected function find_required_missing_params( array $parameters ): array {
-    $missing_params = [];
-    foreach ( $this->get_required_missing_params() as $required_param ) {
+  protected function find_required_missing_params( array $parameters, bool $is_post = false ): array {
```

Comment:
This also looks kinda like it should be a `private` no? Or is it used in the extended classes?

Reply:
Yup good point, can be private.